tools/test-proxy/Azure.Sdk.Tools.TestProxy/Sanitizers/BodyKeySanitizer.cs

```diff
@@ -53,15 +54,22 @@ public override string SanitizeTextBody(string contentType, string body)
                 // HasValues is false for tokens with children. We will not apply sanitization if that is the case.
                 if (!token.HasValues)
                 {
-                    var replacement = StringSanitizer.SanitizeValue(token.Value<string>(), _newValue, _regexValue, _groupForReplace);
+                    var originalValue = token.Value<string>();
+
+                    var replacement = StringSanitizer.SanitizeValue(originalValue, _newValue, _regexValue, _groupForReplace);
 
                     // this sanitizer should only apply to actual values
                     // if we attempt to apply a regex update to a jtoken that has a more complex type, throw
                     token.Replace(JToken.FromObject(replacement));
+
+                    if(originalValue != replacement)
+                    {
+                        sanitized = true;
+                    }
                 }
             }
-            
-            return JsonConvert.SerializeObject(jsonO, SerializerSettings);
+
+            return sanitized ? JsonConvert.SerializeObject(jsonO, SerializerSettings) : body;
```

Comment:
For some reason I didn't get a notification about this PR. It looks great! The only thing missing is the try/catch in this method.